tools/myst/enc/enc.c

```diff
@@ -566,6 +634,19 @@ static long _enter(void* arg_)
 
         _kargs.shell_mode = shell_mode;
         _kargs.memcheck = memcheck;
+        _get_sgx_config_id(_kargs.config_id);
+        if (have_config)
+        {
+            // for (int i = 0; i < 32; i++) {
+            //     oe_host_printf("%x, %x\n", _kargs.config_id[32 + i],
+            //     config_sha256.data[i]);
+            // }
+            if (memcmp(_kargs.config_id + 32, config_sha256.data, 32) != 0)
+            {
+                fprintf(stderr, "config sha256 doesn't match config_id\n");
+                assert(0);
+            }
```

Comment:
Can we save the `config_sha256.data` (i.e., to the `config_id` in kernel.h, although we may need a better naming) after the successful check?